contracts/cw-ics721-bridge/src/contract.rs

```diff
@@ -193,30 +197,37 @@ fn receive_nft(
 
     OUTGOING_CLASS_TOKEN_TO_CHANNEL.save(
         deps.storage,
-        (class_id.clone(), token_id.clone()),
+        (class.id.clone(), token_id.clone()),
         &msg.channel_id,
     )?;
 
     Ok(Response::default()
         .add_attribute("method", "execute_receive_nft")
         .add_attribute("token_id", token_id)
-        .add_attribute("class_id", class_id)
+        .add_attribute("class_id", class.id)
         .add_attribute("channel_id", msg.channel_id)
         .add_message(ibc_message))
 }
 
 fn callback_mint(
-    deps: Deps,
+    deps: DepsMut,
     class_id: ClassId,
     tokens: Vec<Token>,
     receiver: String,
 ) -> Result<Response, ContractError> {
     let receiver = deps.api.addr_validate(&receiver)?;
-    let cw721_addr = CLASS_ID_TO_NFT_CONTRACT.load(deps.storage, class_id)?;
+    let cw721_addr = CLASS_ID_TO_NFT_CONTRACT.load(deps.storage, class_id.clone())?;
 
     let mint = tokens
         .into_iter()
-        .map(|Token { id, uri, data: _ }| {
+        .map(|Token { id, uri, data }| {
+            // We save the metadata here.
```

Comment:
```suggestion
```

Reply:
i think this is a good call. how about a comment about _why_ you choose to save it here? one good reason is that it means we only iterate over the list once.

Reply:
How about: `// We save token metadata here because it is part of the mint process`